### PR TITLE
Run Math expressions tests in Stan Jenkins tests

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py src/test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary

This adds the expressions tests in the Stan repo Jenkins. These tests should be run here because we want these tests to also run on compiler changes (stanc3 will run Stan tests in the near future).

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Rok Češnovar, Univ. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -159,89 +159,103 @@ pipeline {
...
+                // }
+                stage('Math functions expressions test') {
+                    agent any
                     steps {
                         unstash 'StanSetup'
                         setupCXX()
-                        runTests("src/test/integration", separateMakeStep=false)
+                        script {
+                            withEnv(['PATH+TBB=./lib/stan_math/lib/tbb']) {
+                                try { bat "cd lib/stan_math; runTests.py -j${env.PARALLEL} test/expressions; cd .." }
```
Comment (rok-cesnovar) [inline Jenkinsfile:255]: @serban-nicusor-toptal can I ask you for some help here. What would e the best way of running a Python script from the math folder here? I dont think cd-ing in works.

**Comment by serban-nicusor-toptal** [comment]
Hey, Jenkinsfile has an easy way to execute stuff inside a directory using:

```
dir("lib/stan_math"){
  runTests.py -j${env.PARALLEL} test/expressions
}
```

Let me know if I can help with anything else!

**Comment by serban-nicusor-toptal** [comment]
Not sure if I should commit here but there an extra `}` [here](https://github.com/stan-dev/stan/blob/2e174bb053647806368a5d3e6498e9bbe3aa8548/Jenkinsfile#L259)

**Comment by rok-cesnovar** [comment]
Thanks. Sloppy job on my end :)

**Comment by serban-nicusor-toptal** [comment]
Looks like `try-catch` needs to reside inside a `script {}` block, I've committed the change if that's okay.

**Comment by rok-cesnovar** [comment]
Of course its okay. Thanks!

File: Jenkinsfile [hunk 2]
```diff
@@ -245,6 +245,25 @@ pipeline {
...
+                        unstash 'StanSetup'
+                        setupCXX()
+                        script {
+                            dir("lib/stan_math/") {
+                                withEnv(['PATH+TBB=./lib/tbb']) {           
+                                    try { sh "./runTests.py -j${env.PARALLEL} test/expressions" }
+                                    finally { junit 'test/**/*.xml' }
+                                }
+                                withEnv(['PATH+TBB=./lib/tbb']) {           
+                                    sh "python ./test/expressions/test_expression_testing_framework.py"
```
Comment (rok-cesnovar) [inline Jenkinsfile:260]: @t4c1 can you just verify once more that line 256 and 260 are the correct commands? Other stuff is generic Jenkins stuff c/p from other Jenkins stages.

**Pull request merged**